ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp

```diff
@@ -101,9 +101,14 @@ std::unique_ptr<Process> createHydroMechanicsProcess(
             variable_p->getNumberOfComponents());
     }
 
-    // Constitutive relation.
-    auto material =
-        MaterialLib::Solids::createConstitutiveRelation<DisplacementDim>(
+    auto const material_ids =
+        mesh.getProperties().existsPropertyVector<int>("MaterialIDs")
+            ? mesh.getProperties().getPropertyVector<int>("MaterialIDs")
+            : nullptr;
+
+    // Constitutive relations.
+    auto solid_materials =
+        MaterialLib::Solids::createConstitutiveRelations<DisplacementDim>(
```

Comment:
One could collect `material_ids` and `solid_materials` in single class and add the above `select...` function to it...